Data-Structures/Queue/Queue.js

```diff
@@ -1,56 +1,121 @@
 /* Queue
 * A Queue is a data structure that allows you to add an element to the end of
-* a list and remove the item at the front. A queue follows a "First In First Out"
-* system, where the first item to enter the queue is the first to be removed. This
-* implementation uses an array to store the queue.
+* a list and remove the item at the front. A queue follows a FIFO (First In First Out)
+* system, where the first item to enter the queue is the first to be removed,
+* All these operation complexities are O(1).
+* This implementation following the linked list structure.
 */
 
-// Functions: enqueue, dequeue, peek, view, length, empty
 class Queue {
-  // constructor
-  constructor () {
-    // This is the array representation of the queue
-    this.queue = []
+  #size
+
+  constructor (capacity) {
+    if (capacity < 1) {
+      throw new Error('Capacity is not valid')
```

Comment:
Why not? Empty queues should be possible; furthermore capacity shouldn't be mandatory.